expression/builtin_json_vec.go

```diff
@@ -22,6 +22,83 @@ import (
 	"github.com/pingcap/tipb/go-tipb"
 )
 
+func vecJSONModify(b *baseBuiltinFunc, input *chunk.Chunk, result *chunk.Column, mt json.ModifyType) error {
+	nr := input.NumRows()
+	jsonBuf, err := b.bufAllocator.get(types.ETJson, nr)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(jsonBuf)
+	if err := b.args[0].VecEvalJSON(b.ctx, input, jsonBuf); err != nil {
+		return err
+	}
+
+	strBufs := make([]*chunk.Column, (len(b.args)-1)/2)
+	for i := 0; i < (len(b.args)-1)/2; i++ {
+		strBufs[i], err = b.bufAllocator.get(types.ETString, nr)
+		if err != nil {
+			return err
+		}
+		defer b.bufAllocator.put(strBufs[i])
+		if err := b.args[i*2+1].VecEvalString(b.ctx, input, strBufs[i]); err != nil {
+			return err
+		}
+	}
+	valueBufs := make([]*chunk.Column, (len(b.args)-1)/2+1)
+	for i := 0; i < (len(b.args)-1)/2; i++ {
+		valueBufs[i], err = b.bufAllocator.get(types.ETJson, nr)
+		if err != nil {
+			return err
+		}
+		defer b.bufAllocator.put(valueBufs[i])
+		if err := b.args[(i+1)*2].VecEvalJSON(b.ctx, input, valueBufs[i]); err != nil {
+			return err
+		}
+	}
+	result.ReserveJSON(nr)
+	for i := 0; i < nr; i++ {
+		if jsonBuf.IsNull(i) {
+			result.AppendNull()
+			continue
+		}
+		pathExprs := make([]json.PathExpression, 0, (len(b.args)-1)/2+1)
+		values := make([]json.BinaryJSON, 0, (len(b.args)-1)/2+1)
+		var pathExpr json.PathExpression
+		isNull := false
+		for j := 1; j < len(b.args); j++ {
```

Comment:
I think we can split this loop into
` for j := 1; i < len(b.args); j += 2` and 
` for j := 2; j < len(b.args); j += 2` to make it more readable.